### PR TITLE
IIS Rewrite

### PR DESCRIPTION
**Pull request by cyberious**
This is a merge of a rewrite that I had done back in Sept along with another person, I did this on Hack day back in Sept and no movement has come of it from PuppetLabs

### CODE REVIEW

**Comment by juniorsysadmin** [comment]
@cyberious I'm not a user of this module, but the fact that you have got rid of a bunch of execs sounds good already. We um recently became a bit more strict on what fails a test after a modulesync of this module - did you have time to fix the failing tests?

**Comment by cyberious** [comment]
Yeah I am trying to get back to fix the tests, the rebase was a bit of a problem but should have time soonish.

**Comment by cyberious** [comment]
Also most of it is actually deleting manifests and tests that are not needed.

File: LICENSE [hunk 1]
```diff
@@ -1,21 +1,202 @@
...
-The above copyright notice and this permission notice shall be
-included in all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
-IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
-TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
-SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```
Comment (igalic) [inline LICENSE]: uhm… how exactly is it possible that we're re-licensing this, huh?

File: README.md [hunk 2]
```diff
@@ -1,160 +1,169 @@
...
+The application pool which should contain the site. Default: `DefaultAppPool`
+
+* `host_header`<br />
+A host header that should apply to the site. Set to `false` to maintain
+no host header.
+
+* `protocol`<br />
+The protocol for the site. Default `http`
+
+* `ip`<br />
```
Comment (petems) [inline README.md]: These br's seem unnecessary in Markdown?

Comment (igalic) [inline README.md]: i guess they're easier to read (see) than adding spaces at the end.

**Comment by igalic** [comment]
@cyberious other than the license & copyright change, i'm all for this change, but perhaps we should preserve the defines as backwards compatibility shims (shams)?

**Comment by petems** [comment]
This is awesome btw, I'm assuming this prevents the current issue where the checking of if the IIS features are enabled takes a bunch of powershell commands flashing over and over? :+1:

File: LICENSE (hunk 1 above)
Comment (cyberious) [inline LICENSE]: Yeah shouldn't be. Will review and revert that change.

File: README.md (hunk 2 above)
Comment (cyberious) [inline README.md]: No editor inserted them probably.

**Comment by cyberious** [comment]
@petems it should take care of the issue as we have a resource that collects them all initially and if in the correct state we only have one call instead of 1+ per site.

**Comment by petems** [comment]
@igalic Can we merge this then release a major bump to 3.0.0? The performance increases from this are pretty amazing in the tests I did at a customer :smile:

**Comment by jyaworski** [comment]
@cyberious please rebase and/or squash. I would be in favor of merging this. I would imagine Windows environments aren't on 3.x as much as Linux is (me. :frowning:)

**Comment by igalic** [comment]
that looks rebased to me
@jyaworski and i'm fine with the commits as they are

**Comment by igalic** [comment]
hey folks, what's the status of this patchset?
@petems if @cyberious remains busy and unresponsive, would you wanna take over?

**Comment by petems** [comment]
@igalic: Sure, I can rebased and I think we're good to go. I don't want to get too stuck in limbo with the other PR's being blocked by this one, I'd rather click Da Button and clean up as part of a 3.0 release :dancer:

**Comment by petems** [comment]
This is now superceeded by #93, thanks to @cyberious, @WhatsARanjit for their awesome work! 🎆

**Pull request closed without merging**